wrapper/src/main/java/software/amazon/jdbc/PluginServiceImpl.java

```diff
@@ -365,18 +384,37 @@ protected EnumSet<NodeChangeOptions> compare(
   }
 
   @Override
-  public List<HostSpec> getHosts() {
+  public List<HostSpec> getAllHosts() {
     return this.hosts;
   }
 
+  @Override
+  public List<HostSpec> getHosts() {
+    CustomEndpointInfo customEndpointInfo =
+        this.getInfo(this.initialConnectionHostSpec.getHost(), CustomEndpointInfo.class, true);
+    if (customEndpointInfo == null) {
+      return this.hosts;
+    }
+
+    if (STATIC_LIST.equals(customEndpointInfo.getMemberListType())) {
+      return this.hosts.stream()
+          .filter((hostSpec -> customEndpointInfo.getStaticMembers().contains(hostSpec.getHostId())))
```

Comment:
Search isn't efficient since customEndpointInfo.getStaticMembers() is List<String>. It's better to use Set<String>.